apiserver/charms.go

```diff
@@ -241,39 +242,39 @@ func (h *charmsHandler) processPost(r *http.Request, st *state.State) (*charm.UR
 		// There's currently no other time where it makes sense
 		// to accept repository charms through this endpoint.
 		if isImporting, err := modelIsImporting(st); err != nil {
-			return nil, errors.Trace(err)
+			return "", errors.Trace(err)
 		} else if !isImporting {
-			return nil, errors.New("charms may only be uploaded during model migration import")
+			return "", errors.New("charms may only be uploaded during model migration import")
 		}
 	}
 
 	series := query.Get("series")
```

Comment:
Can we add a TODO to drop this when we no longer support migrating from controllers which send it

Reply:
In fact, this whole handler should be dropped when that happens